mongo/options/clientencryptionoptions.go

```diff
@@ -43,7 +82,47 @@ func MergeClientEncryptionOptions(opts ...*ClientEncryptionOptions) *ClientEncry
 		if opt.KmsProviders != nil {
 			ceo.KmsProviders = opt.KmsProviders
 		}
+		if opt.TLSConfig != nil {
+			ceo.TLSConfig = opt.TLSConfig
+		}
 	}
 
 	return ceo
 }
+
+// Used by both ClientEncryptionOptions.SetTLSOptions() and AutoEncryptionOptions.SetTLSOptions()
+func applyTLSOptions(tlsOpts map[string]map[string]interface{}, tlsConfigs map[string]*tls.Config) (map[string]*tls.Config, error) {
+	for provider, opts := range tlsOpts {
+		cfg := &tls.Config{MinVersion: tls.VersionTLS12}
+		
+		for cert := range opts {
+			var err error
+			switch cert {
+			case "tlsCertificateKeyFile", "sslClientCertificateKeyFile":
+				clientCertPath, _ := opts[cert].(string)
+				// apply custom key file password if found, otherwise use empty string
+				if keyPwd, found := opts["tlsCertificateKeyFilePassword"].(string); found {
+					_, err = addClientCertFromConcatenatedFile(cfg, clientCertPath, keyPwd)
+				} else if keyPwd, found := opts["sslClientCertificateKeyPassword"].(string); found {
+					_, err = addClientCertFromConcatenatedFile(cfg, clientCertPath, keyPwd)
+				} else {
+					_, err = addClientCertFromConcatenatedFile(cfg, clientCertPath, "")
+				}
+			case "tlsCertificateKeyFilePassword", "sslClientCertificateKeyPassword":
+				continue
+			case "tlsCAFile", "sslCertificateAuthorityFile":
+				CApath, _ := opts[cert].(string)
```

Comment:
```suggestion
				caPath, _ := opts[cert].(string)
```
An odd case, but `caPath` looks a little better than `CApath`. Also, same comment about erroring when the type is not `string`.